### PR TITLE
Add documentation for unit testing open api shcemas

### PR DESCRIPTION
**Pull request by amoriartyCH**
A document has been added to the /docs directory to help
guide developers through unit testing their schema once they have
added it to the chs-delta-api.

A change to pull common unit testing code into a common folder has been made in this PR as well. By 
moving the code into a common shared file, future developers can make use of the functions provided to significantly 
lower the amount of code needed to test their schemas.

### CODE REVIEW

File: docs/unit-testing-a-new-schema.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+tests used to test the schema.
+
+## 2. Unit test structure
+We elected to structure the open API schema unit tests using a `Karate API Scenario` layout. Each unit test will have a 
+`Given`, `When`, `Then` stage and will use sample request bodies and response bodies to test the schema.
+
+All unit tests will follow the same structure:
+
+1. Load in the wanted request body from a text file using the `common.ReadRequestBody()` function
+2. Create a test httpRequest and set its target to your delta's endpoint, and it's request body to the previously read in
```
Comment (ch-tstrickland) [inline docs/unit-testing-a-new-schema.md:23]: it's -> its

File: docs/unit-testing-a-new-schema.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+request body, and finally set the requests headers.
```
Comment (ch-tstrickland) [inline docs/unit-testing-a-new-schema.md:24]: requests -> request's

File: docs/unit-testing-a-new-schema.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+3. Create an instance of the CHValidator and call it's `ValidateRequestAgainstOpenApiSpec` method, passing it your example
```
Comment (ch-tstrickland) [inline docs/unit-testing-a-new-schema.md:25]: it's -> its

File: docs/unit-testing-a-new-schema.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+Finally, create a new Go file ending in `_test` (e.g. `exampleDeltaSchema_test.go`). This file will contain the unit 
+tests used to test the schema.
+
+## 2. Unit test structure
+We elected to structure the open API schema unit tests using a `Karate API Scenario` layout. Each unit test will have a 
+`Given`, `When`, `Then` stage and will use sample request bodies and response bodies to test the schema.
+
+All unit tests will follow the same structure:
+
+1. Load in the wanted request body from a text file using the `common.ReadRequestBody()` function
```
Comment (ch-tstrickland) [inline docs/unit-testing-a-new-schema.md:22]: May be leave out "wanted"? Sort of sounds like "expected".

File: docs/unit-testing-a-new-schema.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+`request`, `open API spec location` and a dummy `contextId`.
+4. The final step of every unit test is to read in your expected response body using the `common.ReadRequstBody()` function
+and using the `common.CompareActualToExpected(actualResponseBody, expectedResponseBody)` to assert that your expected response
+matches the actual response.
+
+Example structure of a unit test
+```go
+func TestSchemaExample(t *testing.T) {
+
+	Convey("Given I want to test the example-delta API schema", t, func() {
```
Comment (ch-tstrickland) [inline docs/unit-testing-a-new-schema.md:35]: I wonder if this code example could be structured to be an example of an actual test that would encourage them to write meaningful text. 
e.g.
Example structure of a unit test for foo-delta (where foo-delta is the delta type, such as officer-delta)
method name TestFooDeltaMissingMandatory(...)

Convey("Given I want to test that missing mandatory fields in a FooDelta are correctly reported"
...
etc.

Comment (amoriartyCH) [reply docs/unit-testing-a-new-schema.md]: implemented

File: docs/unit-testing-a-new-schema.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+tests used to test the schema.
+
+## 2. Unit test structure
+We elected to structure the open API schema unit tests using a `Karate API Scenario` layout. Each unit test will have a 
+`Given`, `When`, `Then` stage and will use sample request bodies and response bodies to test the schema.
+
+All unit tests will follow the same structure:
+
+1. Load in the example request body from a text file using the `common.ReadRequestBody()` function
+2. Create a test httpRequest and set its target to your delta's endpoint, and its request's body to the previously read in
```
Comment (faisal-amin82) [inline docs/unit-testing-a-new-schema.md:23]: Wrongly placed `'` in `request(s)`

File: docs/unit-testing-a-new-schema.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+request's body, and finally set the requests headers.
```
Comment (faisal-amin82) [inline docs/unit-testing-a-new-schema.md:24]: Same as above

File: docs/unit-testing-a-new-schema.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+## 1. Where to add your unit tests
+Move into the `/validation/schema_testing` directory and create a new directory to contain your unit tests. Inside of your
+new directory, create a set `/request_bodies` and `/response_bodies` directories. These will contain your sample request 
+and response bodies used to unit test the schema.
+
+Finally, create a new Go file ending in `_test` (e.g. `exampleDeltaSchema_test.go`). This file will contain the unit 
+tests used to test the schema.
+
+## 2. Unit test structure
+We elected to structure the open API schema unit tests using a `Karate API Scenario` layout. Each unit test will have a
```
Comment (faisal-amin82) [inline docs/unit-testing-a-new-schema.md:17]: As Tom suggested previously, OpenApi is one word

Comment (amoriartyCH) [reply docs/unit-testing-a-new-schema.md]: Changed this

**Review by faisal-amin82 — APPROVED** [review]

File: docs/unit-testing-a-new-schema.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+tests used to test the schema.
+
+## 2. Unit test structure
+We elected to structure the openAPI schema unit tests using a `Karate API Scenario` layout. Each unit test will have a 
+`Given`, `When`, `Then` stage and will use sample request bodies and response bodies to test the schema.
+
+All unit tests will follow the same structure:
+
+1. Load in the example request body from a text file using the `common.ReadRequestBody()` function
+2. Create a test httpRequest and set its target to your delta's endpoint, and its request's body to the previously read in
```
Comment (ch-tstrickland) [inline docs/unit-testing-a-new-schema.md:23]: I had to read this a couple of times for it to scan. Perhaps:
`2. Create and populate a test httpRequest:`
`a. Set its target to your delta's endpoint`
`b. Set its body to that loaded in step (1)`
`c. Set its headers`

The way to achieve the above in GitHub Markdown is just by indenting the subheading numbers by three spaces - I tried to quote it this way, but GitHub ate the spaces in my comment.

**Review by ch-tstrickland — COMMENTED** [review]
Looks good except for one minor wording comment.

File: docs/unit-testing-a-new-schema.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+## 2. Unit test structure
+We elected to structure the openAPI schema unit tests using a `Karate API Scenario` layout. Each unit test will have a 
+`Given`, `When`, `Then` stage and will use sample request bodies and response bodies to test the schema.
+
+All unit tests will follow the same structure:
+
+1. Load in the example request body from a text file using the `common.ReadRequestBody()` function
+2. Create and populate a test httpRequest 
+    1. Set its target to your delta's endpoint
+    2. Set it's body to that previously loaded in step 1
```
Comment (ch-tstrickland) [inline docs/unit-testing-a-new-schema.md:25]: it's -> its

File: docs/unit-testing-a-new-schema.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    3. Set the request's headers using the provided `SetHeaders` function in `/validation/schema_testing/common/schemaUnitTesting.go`
```
Comment (ch-tstrickland) [inline docs/unit-testing-a-new-schema.md:26]: Set the request's
->
Set its

(for consistency in the list

**Review by ch-tstrickland — APPROVED** [review]

**Pull request merged**